src/utils.ts

```diff
@@ -505,6 +527,155 @@ export function isNodeModule(path: string) {
+      break
+    }
+
+    // nonNull(list(Type))
+    if (isNexusNonNullTypeDef(namedType) && isNexusListTypeDef(namedType.ofType)) {
+      wrapping.unshift({ type: 'List', nullable: false })
+      namedType = namedType.ofType.ofType
+    }
+
+    // nonNull(Type)
+    if (isNexusNonNullTypeDef(namedType) && isWrappedType(namedType.ofType)) {
+      wrapping.unshift({ type: 'WrappedType', nullable: false })
+      namedType = namedType.ofType
+      break
+    }
+
+    // list(...)
+    if (isNexusListTypeDef(namedType)) {
+      wrapping.unshift({ type: 'List', nullable: !nonNullDefault })
+      namedType = namedType.ofType
+    }
+
+    // Type
+    if (!isNexusWrappingType(namedType)) {
+      wrapping.unshift({ type: 'WrappedType', nullable: !nonNullDefault })
+    }
+  }
+
+  return { namedType, wrapping }
+}
+
+export function wrapType(
+  nexusDef: AllNexusOutputTypeDefs | string,
+  baseType: GraphQLNamedType,
+  nonNullDefault: boolean
+): GraphQLOutputType
+export function wrapType(
+  nexusDef: AllNexusInputTypeDefs | string,
+  baseType: GraphQLNamedType,
+  nonNullDefault: boolean
+): GraphQLInputType
+export function wrapType(
+  nexusDef: AllNexusTypeDefs | string,
+  baseType: GraphQLNamedType,
+  nonNullDefault: boolean
+): GraphQLOutputType | GraphQLInputType {
+  const { wrapping } = unwrapNexusDef(nexusDef, nonNullDefault)
+
+  return wrapping.reduce<GraphQLType>((type, kind) => {
+    if (kind.type === 'WrappedType' && kind.nullable === false) {
+      return GraphQLNonNull(type)
+    }
+
+    if (kind.type === 'List' && kind.nullable === true) {
+      return GraphQLList(type)
+    }
+
+    if (kind.type === 'List' && kind.nullable === false) {
+      return GraphQLNonNull(GraphQLList(type))
+    }
+
+    return type
+  }, baseType)
+}
+
+export function wrapAsNexusType(
+  baseType: AllNexusNamedTypeDefs | string,
+  wrapping: NexusWrapKind[]
+): AllNexusOutputTypeDefs
+export function wrapAsNexusType(
+  baseType: AllNexusNamedTypeDefs | string,
+  wrapping: NexusWrapKind[]
+): AllNexusInputTypeDefs
+export function wrapAsNexusType(
+  baseType: AllNexusNamedTypeDefs | string,
+  wrapping: NexusWrapKind[]
+): AllNexusTypeDefs {
+  return wrapping.reduce<any>((type, kind) => {
+    if (kind.type === 'WrappedType' && kind.nullable === false) {
+      return nonNull(type)
+    }
+
+    if (kind.type === 'List' && kind.nullable === true) {
+      return list(type)
+    }
+
+    if (kind.type === 'List' && kind.nullable === false) {
+      return nonNull(list(type))
+    }
+
+    return type
+  }, baseType)
+}
+
+export function getNexusNamedArgDef(argDef: AllNexusArgsDefs | AllInputTypes): AllNexusNamedArgsDefs {
+  if (typeof argDef === 'string') {
+    return argDef
+  }
+
+  let namedType = argDef
+
+  while (isNexusWrappingType(namedType)) {
+    namedType = namedType.ofType
+  }
+
+  return namedType
+}
+
+export function getNexusNamedType(type: AllNexusTypeDefs | string): AllNexusNamedTypeDefs | string {
```

Comment:
```suggestion
/**
 * Strip away list null and nonNull types recursively until another type like an enum or object  
(so-called "named types") is found, then return.
 */
export function unwrapUntilFirstNamedOutputType(type: AllNexusTypeDefs | string): AllNexusNamedTypeDefs | string {
```

Reply:
This named after the `getNamedType()` function from `graphql-js` which achieves the same thing. It's a super commonly used function in all GraphQL tools so I'd personally prefer keeping it named that way

Reply:
Oh I see, fair enough yeah.